source/Nevermore/Advanced/ReaderStrategies/ArbitraryClasses/ArbitraryClassReaderStrategy.cs

```diff
@@ -81,18 +78,31 @@ public bool CanRead(Type type)
         //         result.LastName = reader.GetString(1);
         //         return result;
         //     }
+        //
+        //   -- OR --
+        //
+        //     (DbDataReader reader, ArbitraryClassReaderContext context) =>
+        //     {
+        //         var result = new Person(reader.GetString(0), reader.GetString(1));
+        //         return result;
+        //     }
         CompiledExpression<ArbitraryClassReaderFunc<TRecord>> Compile<TRecord>(IDataRecord record)
         {
             // To make it fast - as fast as if we wrote it by hand - we generate and compile C# expression trees for
             // each property on the class, and one to call the constructor.
-            
-            // Find the parameter-less constructor
+
             var constructors = typeof(TRecord).GetConstructors();
-            var defaultConstructor = constructors.FirstOrDefault(p => p.GetParameters().Length == 0) ?? throw new InvalidOperationException("When reading query results to a class, the class must provide a default constructor with no parameters.");
-            
-            // Create fast setters for all properties on the type
-            var properties = typeof(TRecord).GetProperties(BindingFlags.Public | BindingFlags.Instance | BindingFlags.SetProperty);
-            
+            // Find the parameter-less constructor
```

Comment:
`GetConstructors` will only find public constructors. Should we consider specifying BindingFlags to find any protected/private constructors too? (I think if we did, it'd be good to sort them such that public constructors are matched first). Probably on balance though best left alone?

Reply:
I agree it'd be handy, but status-quo is fine for now.